### PR TITLE
Fix started notebooks disappear issue

### PR DESCRIPTION
**Pull request by ppadti**
Closes: [RHOAIENG-7660](https://issues.redhat.com/browse/RHOAIENG-7660)

## Description
This PR aims to fix the started notebook disappear problem. 

## How Has This Been Tested?
1. Make sure the notebook culling is disabled. (delete the `notebook-controller-culler-config` ConfigMap)
2. Login as a cluster-admin
3. Navigate to the Jupyter tile & select the admin tab
4. Start another user's notebook
5. Check the Administration tab, make sure the started notebook is still visible with last-activity: 'Just now' and kebab action to stop the server.

## Test Impact
Added unit test and cypress test.
Screenshot from 2024-07-02 15-36-30

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by gitdallas — APPROVED** [review]
tested locally and i'm seeing "just now" with the fix in place 
image

File: frontend/src/pages/notebookController/screens/admin/__tests__/useCheckForAllowedUsers.spec.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}));
+
+jest.mock('~/pages/notebookController/useNamespaces', () => () => ({
+  notebookNamespace: 'test-project',
+  dashboardNamespace: 'opendatahub',
+}));
+
+const getAllowedUsersMock = jest.mocked(getAllowedUsers);
+
+describe('useCheckForAllowedUsers', () => {
```
Comment (gitdallas) [inline frontend/src/pages/notebookController/screens/admin/__tests__/useCheckForAllowedUsers.spec.ts:18]: should we have a cypress test to check for "Just now"?

**Review by mturley — COMMENTED** [review]
Code LGTM, but I'm having trouble reaching the Jupyter admin panel to test this. On both localhost (main and this branch) using the modelregistry-ui cluster and on the ui-shared cluster's prod dashboard, when I go to the Administration tab of that page I get an infinite loading spinner. ~~@gitdallas what cluster did you use to repro/test this?~~ (edit: Dallas sent me creds for @ppadti 's cluster)

**Comment by gitdallas** [comment]
@mturley 
cluster link: https://console-openshift-console.apps.pushpa.dev.datahub.redhat.com/ 
username: clusteradminuser1 
password: developer

**Comment by mturley** [comment]
Now trying to test this on Pushpa's cluster and attempting to start the notebook servers for regularuser1 or user-2 are giving me this error (sorry...)

Screenshot 2024-07-11 at 7 12 54 PM

is this happening on your end @ppadti @gitdallas ?

**Comment by ppadti** [comment]
> Now trying to test this on Pushpa's cluster and attempting to start the notebook servers for regularuser1 or user-2 are giving me this error (sorry...)
> 
> Screenshot 2024-07-11 at 7 12 54 PM
> 
> is this happening on your end @ppadti @gitdallas ?

@mturley  yes, this is happening on my end. Could you please try creating notebook for `regularuser3`

File: frontend/src/pages/notebookController/screens/admin/__tests__/useCheckForAllowedUsers.spec.ts (hunk 1 above)
Comment (ppadti) [reply frontend/src/pages/notebookController/screens/admin/__tests__/useCheckForAllowedUsers.spec.ts:18]: @gitdallas I have added cypress test for this.

**Review by gitdallas — APPROVED** [review]

**Comment by mturley** [comment]
Ok, sorry to keep going around with this @ppadti  ... I can now start the notebook server for regularuser3, but the "last activity" it shows is "8 hours ago", not "Just now".

Screenshot 2024-07-12 at 11 41 57 AM

It seems like maybe this is a backend issue? I traced the data back to the `allowedUsers` network call and I see that the `lastActivity` for this user didn't change when I started the server, it's still `2024-07-12T08:06:32Z` (8 hours ago).

**Comment by ppadti** [comment]
> Ok, sorry to keep going around with this @ppadti ... I can now start the notebook server for regularuser3, but the "last activity" it shows is "8 hours ago", not "Just now".
> 
> Screenshot 2024-07-12 at 11 41 57 AM
> It seems like maybe this is a backend issue? I traced the data back to the `allowedUsers` network call and I see that the `lastActivity` for this user didn't change when I started the server, it's still `2024-07-12T08:06:32Z` (8 hours ago).

@mturley  I am sorry. I think the culling was not disabled properly, not sure what happend. Can you give a try now?

**Comment by mturley** [comment]
@ppadti ah, sorry if I should have disabled that culling as part of my test.

Unfortunately, I'm now seeing the bug -- regularuser3 disappeared from the list after I started their notebook server. I'm running on localhost with `7b4bc5e`, which is the latest commit pushed here. Maybe I'm missing something else?

**Comment by mturley** [comment]
ah-- hang on, I was running `start:dev:ext`, so not including your backend changes. one sec...

**Comment by mturley** [comment]
Excellent, I'm now seeing the correct behavior! Thanks for your patience.

**Review by mturley — APPROVED** [review]
Pulled locally and tested using @ppadti's cluster API with local backend

**Pull request merged**